athena-elasticsearch/src/main/java/com/amazonaws/athena/connectors/elasticsearch/ElasticsearchMetadataHandler.java

```diff
@@ -288,37 +296,41 @@ public GetSplitsResponse doGetSplits(BlockAllocator allocator, GetSplitsRequest
-                for (Integer shardId : shardIds) {
-                    // Every split must have a unique location if we wish to spill to avoid failures
-                    SpillLocation spillLocation = makeSpillLocation(request);
-                    // Create a new split (added to the splits set) that includes the domain and endpoint, and
-                    // shard information (to be used later by the Record Handler).
-                    splits.add(new Split(spillLocation, makeEncryptionKey(), ImmutableMap
-                            .of(domain, endpoint, SHARD_KEY, SHARD_VALUE + shardId.toString())));
-                }
-            }
-            catch (IOException error) {
-                throw new RuntimeException("Error retrieving shard-health information: " + error.getMessage(), error);
-            }
+            // We send index request in case the table name is a data stream, a data stream can contains multiple indices which are created by ES
+            // For non data stream, index name is same as table name
+            GetIndexResponse indexResponse = client.indices().get(new GetIndexRequest(request.getTableName().getTableName()), RequestOptions.DEFAULT);
+
+            Set<Split> splits = Arrays.stream(indexResponse.getIndices())
+                    .map(index -> getShardsIDsFromES(client, request.getCatalogName(), index) // get all shards for an index.
+                            .stream()
+                            .map(shardId -> new Split(makeSpillLocation(request), makeEncryptionKey(), ImmutableMap.of(domain, endpoint, SHARD_KEY, SHARD_VALUE + shardId.toString(), INDEX_KEY, index))) // make split for each (index + shardId) combination
+                    )
+                    .reduce(Stream::concat)
+                    .orElseThrow(() -> new RuntimeException(String.format("failed to generate splits for catalog: %s, table name: %s", request.getCatalogName(), request.getTableName())))
+                    .collect(Collectors.toSet());
+
+            return new GetSplitsResponse(request.getCatalogName(), splits);
         }
-        catch (RuntimeException error) {
+        catch (RuntimeException | IOException error) {
             throw new RuntimeException("Error trying to generate splits for index (" +
-                    index + "): " + error.getMessage(), error);
+                    request.getCatalogName() + "): " + error.getMessage(), error);
         }
+    }
 
-        return new GetSplitsResponse(request.getCatalogName(), splits);
+    private Set<Integer> getShardsIDsFromES(AwsRestHighLevelClient client, String catalogName, String index)
+    {
+        try {
+            return client.getShardIds(index, queryTimeout);
+        }
+        catch (RuntimeException | IOException error) {
+            throw new RuntimeException(String.format("Error trying to generate splits for catalog: %s, index: %s, error message: %s", catalogName, index, error.getMessage()), error);
+        }
```

Comment:
Why are you try catching and rethrowing here when the caller is again going to try catch?

You can just declare that this method throws IOException and then just let the caller worry about it.